### PR TITLE
Add FP16

### PR DESCRIPTION
**Pull request by xsacha**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html)
  step by step carefully. **[Yes]**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/xsacha/hunter-h4uji/builds/23154389
  * https://travis-ci.org/xsacha/hunter/builds/507832401

* I have submitted CI configs to https://github.com/ingenue/hunter targeting `pkg.template` branch,
  see this merged pull request https://github.com/ingenue/hunter/pull/345

* My change will resolve this "New package" request: https://github.com/ruslo/hunter/issues/1783

---

### CODE REVIEW

File: cmake/projects/FP16/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+include(hunter_pick_scheme)
+include(hunter_cmake_args)
+
+hunter_add_version(
+    PACKAGE_NAME
+    FP16
+    VERSION
+    0.0.0-febbb1c-p1
+    URL
+    "https://github.com/hunter-packages/FP16/archive/v0.0.0-febbb1c-p0.tar.gz"
```
Comment (ruslo) [inline cmake/projects/FP16/hunter.cmake]: > https://travis-ci.org/xsacha/hunter/jobs/507674728

> -- Using src='https://github.com/xsacha/FP16/archive/v0.0.0-febbb1c-p1.tar.gz'

You're testing and submitting different versions...

File: cmake/projects/FP16/hunter.cmake [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+include(hunter_cacheable)
+include(hunter_download)
+include(hunter_pick_scheme)
+include(hunter_cmake_args)
+
+hunter_add_version(
+    PACKAGE_NAME
+    FP16
+    VERSION
+    0.0.0-febbb1c-p1
```
Comment (ruslo) [inline cmake/projects/FP16/hunter.cmake]: Should be `0.0.0-febbb1c-p0` instead of `0.0.0-febbb1c-p1`.

Comment (xsacha) [reply cmake/projects/FP16/hunter.cmake]: Sorry, was a typo when I created the release in my branch. Forgot to fix.

**Comment by xsacha** [comment]
Thanks. There is the docs.

File: cmake/projects/FP16/hunter.cmake (hunk 1 above)
Comment (xsacha) [reply cmake/projects/FP16/hunter.cmake]: Updated tests. It was same code

**Comment by ruslo** [comment]
Testing:
- https://ci.appveyor.com/project/ingenue/hunter/builds/23181465
- https://travis-ci.org/ingenue/hunter/builds/508262893?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
- https://ci.appveyor.com/project/ingenue/hunter/builds/23181806
- https://travis-ci.org/ingenue/hunter/builds/508268317?utm_source=github_status&utm_medium=notification

**Pull request merged**